### PR TITLE
Update Brew-Livecheck.md: add examples for modify version information

### PR DESCRIPTION
**Pull request by muescha**
## Changes

Add examples for modifying version information, as there is no information about this in the documentation.

## Sources:
- https://github.com/homebrew/brew/issues/10866#issuecomment-804090944
- https://github.com/Homebrew/homebrew-cask/blob/5d2fa32d67e525794d765f2b7927159b447a1d41/Casks/e/evernote.rb#L41-L48
- GitHub code search for: [`"strategy :electron_builder do"`](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+%22strategy+%3Aelectron_builder+do%22&type=code)

## Checks
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

File: docs/Brew-Livecheck.md [hunk 1]
```diff
@@ -268,6 +268,34 @@ livecheck do
 end
 ```
 
+If you need to modify the version, you can access the yaml hash in the strategy block like so:
```
Comment (samford) [inline docs/Brew-Livecheck.md:271]:
```suggestion
If you need to modify the version, you can access the YAML hash in the `strategy` block like so:
```

File: docs/Brew-Livecheck.md [hunk 2]
```diff
@@ -268,6 +268,34 @@ livecheck do
...
+```ruby
+livecheck do
+  url "https://example.org/my-app/latest-mac.yml"
+  strategy :electron_builder do |yaml|
+    yaml["version"]&.gsub(/\D/, "")
+  end
+end
+```
+
+Or a more complex one:
```
Comment (samford) [inline docs/Brew-Livecheck.md:282]:
```suggestion
Similarly, you can work with the `files` array like this:
```

File: docs/Brew-Livecheck.md [hunk 3]
```diff
@@ -268,6 +268,34 @@ livecheck do
...
+```ruby
+livecheck do
+  url "https://example.org/my-app/latest-mac.yml"
+  regex(/MyApp[._-](\d+(?:\.\d+)+)-mac-ddl-stage-(\d+(?:\.\d+)*)-([0-9a-f]+)\.dmg/i)
+  strategy :electron_builder do |yaml, regex|
+    yaml["files"]&.map do |file|
+      match = file["url"]&.match(regex)
+      next if match.blank?
+
+      "#{match[1]},#{match[2]},#{match[3]}"
```
Comment (samford) [inline docs/Brew-Livecheck.md:293]:
```suggestion
  regex(/MyApp[._-]v?(\d+(?:\.\d+)+)-(\h+)\.dmg/i)
  strategy :electron_builder do |yaml, regex|
    yaml["files"]&.map do |file|
      match = file["url"]&.match(regex)
      next if match.blank?

      "#{match[1]},#{match[2]}"
```

I try to keep the examples as simple as possible while still explaining the desired point. I think we can simplify the regex and `strategy` block here while still explaining the same idea.

This also makes a few tweaks to bring the regex in line with prevailing standards.

-----

For what it's worth, outside of the context of explaining this idea, we also sometimes do something like this in practice to create CSV version values:

```ruby
livecheck do
  url "https://example.org/my-app/latest-mac.yml"
  regex(/MyApp[._-]v?(\d+(?:\.\d+)+-\h+)\.dmg/i)
  strategy :electron_builder do |yaml, regex|
    yaml["files"]&.map do |file|
      match = file["url"]&.match(regex)
      next if match.blank?

      match.tr("-", ",")
   end
  end
end
```

**Review by samford — COMMENTED** [review]
Just a few tweaks and I think this is good.

[I still need to go through this documentation and bring it up to date (and I have some other changes planned to hopefully make this more maintainable) but I need to create a PR for a functional change to `#regex` first.]

File: docs/Brew-Livecheck.md (hunk 3 above)
Comment (muescha) [reply docs/Brew-Livecheck.md:293]: should the new CSV example also included?

**Comment by muescha** [comment]
Should the new example from [this comment](https://github.com/Homebrew/brew/pull/17798#discussion_r1683068829) also be included? 

I am curious about the advantages of having a CSV version of the values.

**Comment by samford** [comment]
> Should the new example from [this comment](https://github.com/Homebrew/brew/pull/17798#discussion_r1683068829) also be included?

The goal is to explain the `files` array and the one example does that, so no need for two.

The other example I mentioned applies to `strategy` blocks in general, not `ElectronBuilder` specifically. There are a number of variations to handle certain scenarios and we don't document them all here (as it would end up being very lengthy and there's been some pushback about documentation doing too much "hand-holding").

Instead, I've tried to present some general ideas and contributors can look through existing `livecheck` blocks to identify contextually-appropriate approaches like the one I mentioned (granted, some `livecheck` blocks do things that go against prevailing standards and shouldn't be copied). Past that, I also give feedback/suggestions when reviewing PRs to help contributors use common patterns/approaches.

-----

> I am curious about the advantages of having a CSV version of the values.

It's just a necessity for certain casks. Some have a `url` with a path and/or filename that requires more than just a version string, so we handle this with a CSV `version` value. For example, you may have a URL that has a hash in the path like `url "https://example.com/1234abcd/Example-1.2.3.zip"`. In this scenario, we have to do something like this with the `version`, `url`, and `livecheck` block:

```ruby
version "1.2.3,1234abcd"

url "https://example.com/#{version.csv.second}/Example-#{version.csv.first}.zip"

livecheck do
  url "https://example.com/downloads"
  regex(%r{href=.*?/(\h+)/Example[._-]v?(\d+(?:\.\d+)+)\.zip}i)
  strategy do |page, regex|
    page.scan(regex).map { |match| "#{match[1]},#{match[0]}" }
  end
end
```

This type of setup makes it easy to update the cask to a new version (e.g., using `brew bump-cask-pr`), instead of hardcoding the hash and requiring users to edit the cask file to update the version.

**Review by samford — APPROVED** [review]
Thanks, @muescha!

**Comment by muescha** [comment]
Thanks :)

**Pull request merged**